src/ProjectSystemTools/BuildLogging/Model/Frontend/FrontendBuildTableDataSource.cs

```diff
@@ -143,21 +142,14 @@ public ITableEntriesSnapshot GetSnapshot(int versionNumber)
             return null;
         }
 
-        public string GetLogForBuild(int buildID)
+        public async Task<string> GetLogForBuild(int buildID)
         {
-            string result;
-            result = ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
-            {
-                string result = await _loggerService.GetLogForBuild(buildID);
-                return result;
-            });
-            return result;
+            return await _loggerService.GetLogForBuild(buildID);
         }
 
-        private void UpdateEntries()
+        private async void UpdateEntries()
```

Comment:
`async void` is to be avoided. It makes it impossible for the caller to block on the async work.